src/reflect/type.go

```diff
@@ -430,31 +447,45 @@ type structField struct {
 // Equivalent to (go/types.Type).Underlying(): if this is a named type return
 // the underlying type, else just return the type itself.
 func (t *rawType) underlying() *rawType {
-	if t.meta&flagNamed != 0 {
+	if t.isNamed() {
 		return (*elemType)(unsafe.Pointer(t)).elem
```

Comment:
Ok.  I was assuming that because `namedType` and `elemType` shared the same initial struct layout this would be safe.

Reply:
You are correct. I was assuming a different struct layout. No need to change anything here.

Reply:
One possible optimization to named types is to use the following struct layout:

```go
type namedType struct {
	rawType
	nlen       uint16
	ptrTo      *rawType
	underlying *rawType
	name       [1]byte
}
```

This saves a single pointer-sized integer on most architectures (32-bit and 64-bit) for each named type by storing `nlen` in the padding that's between `rawType` and `ptrTo`. But on the other hand, this might not be worth the complexity and can also be done at a later time.

(I was assuming this optimization was used here, in which case my original comment would have been correct).